### PR TITLE
Allow editting entity registry

### PR DESCRIPTION
**Pull request by balloob**
Add a config panel page to manage the entity registry.

Requires: https://github.com/home-assistant/home-assistant/pull/20597

In the screenshots, the ones without a name are not currently loaded in HA

image

image

image

### CODE REVIEW

File: src/panels/config/entity_registry/show-dialog-entity-registry-detail.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+}
+
+export const loadEntityRegistryDetailDialog = () =>
+  import(/* webpackChunkName: "entity-registry-detail-dialog" */ "./dialog-entity-registry-detail");
+
+export const showEntityRegistryDetailDialog = (
+  element: HTMLElement,
+  systemLogDetailParams: EntityRegistryDetailDialogParams
+): void => {
+  fireEvent(element, "show-dialog", {
```
Comment (balloob) [inline src/panels/config/entity_registry/show-dialog-entity-registry-detail.ts]: New and improved way of doing dialogs.

CC @bramkragten

File: src/data/entity_registry.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+export interface EntityRegistryEntryUpdateParams {
+  name: string;
+  new_entity_id: string;
+}
+
+export const computeEntityRegistryName = (
+  hass: HomeAssistant,
+  entry: EntityRegistryEntry
+) => {
```
Comment (iantrich) [inline src/data/entity_registry.ts]:
```suggestion
): string | null => {
```

File: src/data/entity_registry.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  if (entry.name) {
+    return entry.name;
+  }
+  const state = hass.states[entry.entity_id];
+  return state ? computeStateName(state) : null;
+};
+
+export const fetchEntityRegistry = (hass: HomeAssistant) =>
```
Comment (iantrich) [inline src/data/entity_registry.ts]:
```suggestion
export const fetchEntityRegistry = (hass: HomeAssistant): Promise<EntityRegistryEntry[]> =>
```

File: src/data/entity_registry.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  hass.callWS<EntityRegistryEntry[]>({ type: "config/entity_registry/list" });
+
+export const updateEntityRegistryEntry = (
+  hass: HomeAssistant,
+  entityId: string,
+  updates: Partial<EntityRegistryEntryUpdateParams>
+) =>
```
Comment (iantrich) [inline src/data/entity_registry.ts]:
```suggestion
): Promise<void> =>
```

File: src/data/entity_registry.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  hass.callWS<EntityRegistryEntry>({
+    type: "config/entity_registry/update",
+    entity_id: entityId,
+    ...updates,
+  });
+
+export const removeEntityRegistryEntry = (
+  hass: HomeAssistant,
+  entityId: string
+) =>
```
Comment (iantrich) [inline src/data/entity_registry.ts]:
```suggestion
): Promise<void> =>
```

File: src/dialogs/more-info/more-info-settings.js [hunk 6]
```diff
@@ -122,12 +123,14 @@ class MoreInfoSettings extends LocalizeMixin(EventsMixin(PolymerElement)) {
 
   async _save() {
```
Comment (iantrich) [inline src/dialogs/more-info/more-info-settings.js]:
```suggestion
  async _save(): Promise<void> {
```

File: src/panels/config/entity_registry/dialog-entity-registry-detail.ts [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+            .disabled=${invalidDomainUpdate || this._submitting}
+          >
+            UPDATE
+          </paper-button>
+        </div>
+      </paper-dialog>
+    `;
+  }
+
+  private _nameChanged(ev: PolymerChangedEvent<string>) {
```
Comment (iantrich) [inline src/panels/config/entity_registry/dialog-entity-registry-detail.ts]:
```suggestion
  private _nameChanged(ev: PolymerChangedEvent<string>): void {
```

File: src/panels/config/entity_registry/dialog-entity-registry-detail.ts [hunk 8]
```diff
@@ -0,0 +1,180 @@
...
+    this._error = undefined;
+    this._name = ev.detail.value;
+  }
+
+  private _entityIdChanged(ev: PolymerChangedEvent<string>) {
```
Comment (iantrich) [inline src/panels/config/entity_registry/dialog-entity-registry-detail.ts]:
```suggestion
  private _entityIdChanged(ev: PolymerChangedEvent<string>): void {
```

File: src/panels/config/entity_registry/dialog-entity-registry-detail.ts [hunk 9]
```diff
@@ -0,0 +1,180 @@
...
+    this._error = undefined;
+    this._entityId = ev.detail.value;
+  }
+
+  private async _updateEntry() {
```
Comment (iantrich) [inline src/panels/config/entity_registry/dialog-entity-registry-detail.ts]:
```suggestion
  private async _updateEntry(): Promise<void> {
```

File: src/panels/config/entity_registry/dialog-entity-registry-detail.ts [hunk 10]
```diff
@@ -0,0 +1,180 @@
...
+        name: this._name,
+        new_entity_id: this._entityId,
+      });
+      this._params = undefined;
+    } catch (err) {
+      this._error = err;
+    }
+  }
+
+  private async _deleteEntry() {
```
Comment (iantrich) [inline src/panels/config/entity_registry/dialog-entity-registry-detail.ts]:
```suggestion
  private async _deleteEntry(): Promise<void> {
```

File: src/panels/config/entity_registry/ha-config-entity-registry.ts [hunk 11]
```diff
@@ -0,0 +1,147 @@
...
+                </paper-item-body>
+              </paper-icon-item>
+            `;
+          })}
+        </paper-card>
+      </hass-subpage>
+    `;
+  }
+
+  protected firstUpdated(changedProps) {
```
Comment (iantrich) [inline src/panels/config/entity_registry/ha-config-entity-registry.ts]:
```suggestion
  protected firstUpdated(changedProps): void {
```

File: src/panels/config/entity_registry/ha-config-entity-registry.ts [hunk 12]
```diff
@@ -0,0 +1,147 @@
...
+    super.firstUpdated(changedProps);
+    this._fetchData();
+    loadEntityRegistryDetailDialog();
+  }
+
+  private async _fetchData() {
```
Comment (iantrich) [inline src/panels/config/entity_registry/ha-config-entity-registry.ts]:
```suggestion
  private async _fetchData(): Promise<void> {
```

File: src/panels/config/entity_registry/ha-config-entity-registry.ts [hunk 13]
```diff
@@ -0,0 +1,147 @@
...
+    this._items = (await fetchEntityRegistry(this.hass!)).sort((ent1, ent2) =>
+      compare(ent1.entity_id, ent2.entity_id)
+    );
+  }
+
+  private _openEditEntry(ev: MouseEvent) {
```
Comment (iantrich) [inline src/panels/config/entity_registry/ha-config-entity-registry.ts]:
```suggestion
  private _openEditEntry(ev: MouseEvent): void {
```

**Comment by iantrich** [comment]
Did this review on my phone, so could very well have missed stuff 😀

File: src/data/entity_registry.ts (hunk 3 above)
Comment (balloob) [reply src/data/entity_registry.ts]: TypeScript is in this case actually smart enough to take the return type form `hass.callWS`

File: src/panels/config/entity_registry/show-dialog-entity-registry-detail.ts (hunk 1 above)
Comment (bramkragten) [reply src/panels/config/entity_registry/show-dialog-entity-registry-detail.ts]: 🎉

File: src/panels/config/entity_registry/dialog-entity-registry-detail.ts [hunk 14]
```diff
@@ -0,0 +1,191 @@
...
+      });
+      this._params = undefined;
+    } catch (err) {
+      this._submitting = false;
+      this._error = err;
+    }
+  }
+
+  private async _deleteEntry(): Promise<void> {
+    this._submitting = true;
```
Comment (bramkragten) [inline src/panels/config/entity_registry/dialog-entity-registry-detail.ts]: Should set to false when done.

Comment (balloob) [reply src/panels/config/entity_registry/dialog-entity-registry-detail.ts]: https://github.com/home-assistant/home-assistant-polymer/pull/2643

**Pull request merged**